src/backends/github/implementation.js

```diff
@@ -136,8 +136,18 @@ export default class GitHub {
       const response = await this.api.persistFiles(null, [mediaFile], options);
       const repo = this.repo || this.getRepoFromResponseUrl(response.url);
       const { value, size, path, fileObj } = mediaFile;
-      const url = `https://raw.githubusercontent.com/${repo}/${this.branch}${path}`;
-      return { id: response.sha, name: value, size: fileObj.size, url, path: trimStart(path, '/') };
+      let url = `https://raw.githubusercontent.com/${repo}/${this.branch}${path}`;
+
+      // Assets uploaded to private repos will need valid access tokens.
+      const files = await this.api.listFiles(this.config.get('media_folder'));
```

Comment:
I think we should only get the file that we are actually wanting, not the entire `media_folder` again. This could be accomplished by making a function essentially like `listFiles`, except for requesting a single file instead of a directory.

Reply:
Also, I do think double-checking the SHA here is a good idea, as we don't want to run into caching issues. When we were constructing the URL manually before, though, we definitely weren't worrying about caching.

Reply:
@tech4him1

> I think we should only get the file that we are actually wanting

I wanted to do that, but I couldn't find any API in the docs that could be used for that. The [Contents API](https://developer.github.com/v3/repos/contents/#get-contents) being used now, if pointed at a file rather than at a directory, will also retrieve the contents of that file. So we'd be transferring the whole image file twice, once just to get the metadata containing its `download_url`, and again later on to show the thumbnail (as it's a different url so it's not cached).

That API also has a 1 Mb limit, so it won't work for larger files, we'd need the [Blobs API](https://developer.github.com/v3/git/blobs/#get-a-blob) to retrieve those files, but that doesn't include the `download_url`.

As far as I can tell, this really seems to be the most direct way to get the tokenized `download_url` currently used to show the thumbnails. Those alternatives are probably useful for #990 though.

Reply:
The only other option I can think of is super hacky - parse the token from the download_url of another image. 99.9% sure we just shouldn't.

So that aside, we should:

1) only take the approach provided here for private repos
2) hurry up and get thumbnail creation in place (#946), and ensure thumbnails are well below 1mb, so we can stop doing this